content/articles/creating-serverless-functions-faas-with-aws-lambda-and-golang/index.md

```diff
@@ -0,0 +1,275 @@
+
+```bash
+
+go get github.com/aws/aws-sdk-go
+
+```
+
+We can now use this code segment to execute the function in the cloud and see the results locally.
+
+```go
+
+package main
+
+ 
+
+import (
+
+ "encoding/json"
+
+ "fmt"
+
+ "os"
+
+ 
+
+ "github.com/aws/aws-sdk-go/aws"
+
+ "github.com/aws/aws-sdk-go/service/lambda"
+
+ "github.com/joho/godotenv"
+
+ 
+
+ "github.com/aws/aws-sdk-go/aws/credentials"
+
+ "github.com/aws/aws-sdk-go/aws/session"
+
+ // "github.com/aws/aws-lambda-go/lambda"
+
+)
+
+ 
+
+type Info struct {
+
+ Firstname string `json:"firstname"`
+
+ Lastname string `json:"lastname"`
+
+ Age int `json:"age"`
+
+}
+
+ 
+
+type Response struct {
+
+ Profile string `json:"profile"`
+
+}
+
+ 
+
+func main() {
+ // Load the environment variable which has the IAM credentials needed to connect to AWS
+ godotenv.Load()
+ // Initialize new aws session
+ sess := session.Must(session.NewSessionWithOptions(session.Options{SharedConfigState: session.SharedConfigEnable}))
+ // Initialize new aws client using IAM credentials
+ client := lambda.New(sess, &aws.Config{Region: aws.String("us-east-1"), Credentials: credentials.NewStaticCredentials(os.Getenv("aws_access_key_id"), os.Getenv("aws_secret_access_key"), "")})
+ // Prepare request parameters
+ request := Info{Firstname: "Oluwatomisin", Lastname: "Bamimore", Age: 16}
+
+ payload, err := json.Marshal(request)
+
+ if err != nil {
+
+ fmt.Println("Error marshalling request")
+
+ os.Exit(0)
+
+ }
+
+ result, err := client.Invoke(&lambda.InvokeInput{FunctionName: aws.String("user-profile"), Payload: payload})
+
+ if err != nil {
+
+ fmt.Println("Error calling user-profile function", err)
+
+ os.Exit(0)
+
+ }
+
+ var resp Response
+
+ err = json.Unmarshal(result.Payload, &resp)
+
+ if err != nil {
+
+ fmt.Println("Error unmarshalling user-profile response")
+
+ os.Exit(0)
+
+ }
+
+ 
+ // Print response from lambda function
+ fmt.Println(resp.Profile)
+
+}
+
+```
+
+You need to create an [IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) user with the correct permissions. Store the `aws_access_key_id` and the `aws_secret_access_key` in a `.env` file in the directory you are currently working in.
+
+![sample-env-file](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/sample-env-file.png)After running the program, AWS will execute the cloud function, and you will get this result.
+
+![sample-program-output](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/sample-program-output.png)
+
+The program passes in some arguments (json format) into the cloud function and returns the result.
+
+### Conclusion
+
+We have successfully deployed and tested our AWS Lambda function. We also tested the Lambda function using an actual client program and the Lambda console on AWS. In other languages, such as Python, JavaScript, and Java, the same mechanism can be used to create serverless functions.
```

Comment:
Add key takeaways from this article.

Add a `Further reading` section that references other links that the reader can read through

Reply:
The key takeaway is clearly stated there. Uploading a function to AWS lambda and testing it right in the cloud and locally.

I have clearly stated this in my conclusion..